consensus/raft/logging.go

```diff
@@ -58,19 +58,38 @@ func (fw *logForwarder) Write(p []byte) (n int, e error) {
 }
 
 func (fw *logForwarder) repeated(t int, msg string) bool {
-	if fw.last == nil {
-		fw.last = make(map[int]*lastMsg)
+	if fw.lastMsgs == nil {
+		fw.lastMsgs = make(map[int][]string)
+		fw.lastTip = make(map[int]time.Time)
+	}
+
+	// We we haven't tipped about repeated log messages
+	// in while, do it and forget the list
+	if time.Now().After(fw.lastTip[t].Add(repeatReset)) {
```

Comment:
Just double checking this: this changes the raft logger to periodically remind the user that raft messages can repeat but does not depend on whether or not a particular message is repeating itself?

Reply:
Why do we have a separate interval for giving notice depending on the loglevel of the message?  It seems like this will lead to unnecessary repeated log messages in the case that the raft logs are giving out more than one log level frequently.  If I am correct that the tip message is decoupled from witnessing actual repeats then maybe it makes more sense to just keep track of a single interval?

Reply:
You're correct it's decoupled. Reason is that the NOTICE message would repeat itself every time that a line repeats itself for the first time. So just taking care of that.

There is one per log level because this code is called with debug lines all the time (even if not printed). If there was one single `lastTip`, debug logs would keep hitting this condition and printing the notice constantly. If they print it as debug, then it might not appear at all when the log level is error.